great_expectations/data_context/data_context.py

```diff
@@ -2443,13 +2450,17 @@ def validations_store_name(self):
     def validations_store(self):
         return self.stores[self.validations_store_name]
 
-    def _compile_evaluation_parameter_dependencies(self):
+    def _compile_evaluation_parameter_dependencies(self, expectation_suite):
         self._evaluation_parameter_dependencies = {}
-        for key in self.expectations_store.list_keys():
-            expectation_suite = self.expectations_store.get(key)
-            if not expectation_suite:
-                continue
 
+        # only if we don't have an expectation suite do we do a key scan
+        if expectation_suite is None:
```

Comment:
Hi @omribromberg - thanks so much for finding and flagging this. This was an unintended consequence. If you've already tracked this down and are up to making a fix, we would greatly appreciate it. I can prioritize reviewing this as soon as it's ready.

Reply:
Also just a heads-up, I believe that #3660 which was recently filed is pointing to the same issue.

Reply:
@talagluck I think these might be different issues but might have the same solution, to solve evaluation parameters of this pattern: `urn:great_expectations:validations:{expectation_suite}:{expectation}.result.unexpected_percent:column={column_name}` which is what I was using, we'll need to go through all expectation suites' dependencies each time the action `StoreEvaluationParametersAction` is called to figure out if the current validation metrics needs to be saved for future use
correct me if I'm wrong, but removing the `expectation_suite` parameter from `_compile_evaluation_parameter_dependencies` would solve this with no apparent new bugs
lemme know if I can open a separate PR for this specific issue

Reply:
Hi @OmriBromberg - thanks for the follow-up.

After some internal conversation, we believe that reverting this change is the right thing to do for the moment. It will possibly lead to another bug being resurfaced, but that bug is an edge case with a work-around (for which some more in-depth fixes are possible), whereas this one feels more urgent.

The fix you are proposing sounds right - it may also require the reversion of tests from this PR. Please do open a PR, and I will take a look once it's ready.

Thanks again!